lib/main.dart

```diff
@@ -1,36 +1,53 @@
 import 'package:flutter/material.dart';
 
-import 'src/presentation/custom_widgets.dart';
+import 'src/config/route/app_routes.dart';
+import 'src/config/theme/app_themes.dart';
+import 'src/core/parameter/init_values.dart';
 import 'src/core/util/ui_constants.dart';
 import 'src/domain/entity/movie.dart';
 
-void main() => runApp(MyApp());
+void main() => runApp(const MyApp());
 
-class MyApp extends StatelessWidget {
-  MyApp({super.key});
+class MyApp extends StatefulWidget {
+  const MyApp({
+    super.key,
+  });
 
+  @override
+  State<MyApp> createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
   final Movie movie = Movie.mockMovie();
 
+  late InitCore _initCore;
+
+  late Routes _routes;
+
+  @override
+  void initState() {
+    super.initState();
+    _routes = Routes();
+    _initCore = InitCore();
+  }
+
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      theme: ThemeData(
-        colorScheme: ColorScheme.fromSeed(
-          seedColor: Constants.customBlueColor,
-          background: Constants.customPurpleColor,
-        ),
-        scaffoldBackgroundColor: Constants.customBlueColor,
-        textTheme: const TextTheme(
-          bodyLarge: TextStyle(color: Colors.white70),
-          bodyMedium: TextStyle(color: Colors.white70),
-          bodySmall: TextStyle(color: Colors.white70),
-        ),
-        appBarTheme: const AppBarTheme(
-          color: Colors.black,
-          elevation: Constants.appBarElevation,
-        ),
-      ),
-      home: const HomePage(),
+    return FutureBuilder<bool>(
+      future: _initCore.initialize(),
+      builder: (context, AsyncSnapshot<bool> snapshot) {
```

Comment:
Add type of parameters, and also you missed a trailing comma

Reply:
Done